4.3 - Remove formula approach, add API and manual pillar method (#2386)
uyuni-project/uyuni-docs: modules/retail/pages/retail-saltboot-redeployment.adoc

+
----
spacecmd api -- -A '$terminal_minion_id,tuning-saltboot,{"saltboot": {"force_redeployment": "True"}}' system.setPillar
----

.Procedure: Forcing a {saltboot} to repartition disk using API call using spacecmd command
. In the console run following command, replace [systemitem]``$terminal_minion_id`` with actual terminal minion id:
+
----
spacecmd api -- -A '$terminal_minion_id,tuning-saltboot,{"saltboot": {"force_repartition": "True"}}' system.setPillar
----

.Procedure: Check {saltboot} tuning options
. In the console run following command, replace [systemitem]``$terminal_minion_id`` with actual terminal minion id:
+
----
spacecmd api -- -A '$terminal_minion_id,tuning-saltboot' system.getPillar
----

[IMPORTANT]
====
Make sure to use [systemitem]``tuning-saltboot`` as pillar category in the API call.
====

[[force.redeployment.using.pillar]]
== Force {saltboot} redeployment by custom pillar

[WARNING]
====
Pillars specified outside of {productname} database cannot be reset automatically. Without manual intervention, terminal will download new image on each reboot.
====

.Procedure: Force a {saltboot} to redeploy image using {saltboot} pillar

> . Create new file named ``/srv/salt/pillar/force_redeploy.sls`` with content:

Reviewer: ```suggestion
. Create new file [filename]``/srv/salt/pillar/force_redeploy.sls`` with content:
```

+
----
saltboot:
  force_redeploy: True

> ----

Reviewer: ```suggestion
----
+
```

. Create new file or update existing file named ``/srv/salt/pillar/top.sls`` with content:
+
----
base:
  '$terminal_minion_id':
    - force_redeploy

> ----

Reviewer: ```suggestion
----
+
```

. Reboot the terminal to pick up the changes.

> . After terminal finishes booting, remove modifications done in ``/srv/salt/pillar/top.sls`` file.

Reviewer: ```suggestion
. After the terminal finishes booting, remove modifications made in [filename]``/srv/salt/pillar/top.sls`` file.
```

If your terminal encounters a problem with the file system or the partition table, you might need to remove the partition table and reformat the terminal.

> .Procedure: Force a {saltboot} to repartition disk using {saltboot} pillar

Reviewer: ```suggestion
.Procedure: Force {saltboot} to repartition disk using {saltboot} pillar
```

> . Create new file named ``/srv/salt/pillar/force_repartition.sls`` with content:

Reviewer: ```suggestion
. Create new file [filename]``/srv/salt/pillar/force_repartition.sls`` with content:
```

+
----
saltboot:
  force_repartition: True

> ----

Reviewer: ```suggestion
----
+
```

. Create new file or update existing file named ``/srv/salt/pillar/top.sls`` with content:
+
----
base:
  '$terminal_minion_id':
    - force_repartition

> ----

Reviewer: ```suggestion
----
+
```

. Reboot the terminal to pick up the changes.

> . After terminal finishes booting, remove modifications done in ``/srv/salt/pillar/top.sls`` file.

Reviewer: ```suggestion
. After the terminal finishes booting, remove modifications made in [filename]``/srv/salt/pillar/top.sls`` file.
```
